sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -4077,19 +4061,18 @@ class AstBuilder extends SqlBaseParserBaseVisitor[AnyRef] with SQLConfHelper wit
     } else {
       None
     }
-    val setDefaultExpression: Option[String] =
-      if (action.defaultExpression != null) {
-        Option(action.defaultExpression()).map(visitDefaultExpression)
-      } else if (action.dropDefault != null) {
-        Some("")
-      } else {
-        None
-      }
-    if (setDefaultExpression.isDefined && !conf.getConf(SQLConf.ENABLE_DEFAULT_COLUMNS)) {
+    val defaultExpression: Option[Expression] = if (action.defaultExpression != null) {
+      Some(visitDefaultExpression(action.defaultExpression))
+    } else if (action.dropDefault != null) {
+      Some(DropDefaultColumnValue)
```

Comment:
We can use `Some(null)` to indicate removing default value but I think this is better.

Reply:
Yes, this is more explicit.